runtime_directory/sandbox_run.ps1

```diff
@@ -32,29 +40,88 @@ function Remove-VmComputeAgent {
     Remove-Item -Path $vmcompute_path
 }
 
-function Extract-LockdownBrowser {
-    & $lockdown_installer /x "`"$lockdown_extract_dir`"" # Dumb installer needs a quoted path, even with no spaces. Also, we have to extract the program before we can even run a silent install.
-    while (!(Test-Path "$lockdown_extract_dir\id.txt")) {
-        # This is the easiest way to detect if the installer is finished extracting
-        Start-Sleep -Seconds 0.2
+function Expand-LockdownBrowser {
+    if ($lockdown_installer.Name -like "LockDownBrowserOEMSetup.exe") {
+        break
+    }
+    else {
+        & $lockdown_installer /x "`"$lockdown_extract_dir`"" # Dumb installer needs a quoted path, even with no spaces. Also, we have to extract the program before we can even run a silent install.
+        while (!(Test-Path "$lockdown_extract_dir\id.txt")) {
+            # This is the easiest way to detect if the installer is finished extracting
+            Start-Sleep -Seconds 0.2
+        }
+        Start-Sleep -Seconds 1
+        Stop-Process -Name *Lockdown* -ErrorAction Ignore
     }
-    Start-Sleep -Seconds 1
-    Stop-Process -Name *Lockdown* -ErrorAction Ignore
 }
 
 function Install-LockdownBrowser {
-    & "$lockdown_extract_dir\setup.exe" /s "/f1$PSScriptRoot\setup.iss" "/f2$PSScriptRoot\..\setup.log" # If we don't give a log file path, this doesn't work
-    Wait-Process -Name "setup"
+    if ($lockdown_installer.Name -like "LockDownBrowserOEMSetup.exe") {
+        & $lockdown_installer /s /r
+        $test = "C:\Program Files (x86)\Respondus\LockDown Browser OEM\"
+        while (!(Test-Path "$test/am.pak")) {
+            # This is the easiest way to detect if the installer is finished extracting
+            Start-Sleep -Seconds 0.2
+        }
+        Start-Sleep -Seconds 1
+        Stop-Process -Name *Setup* -ErrorAction Ignore
+    }
+    else {
+        & "$lockdown_extract_dir\setup.exe" /s "/f1$PSScriptRoot\setup.iss" "/f2$PSScriptRoot\..\setup.log" # If we don't give a log file path, this doesn't work
+        Wait-Process -Name "setup"
+    }
+
 }
 
 function Register-URLProtocol {
-    New-PSDrive -PSProvider registry -Root HKEY_CLASSES_ROOT -Name HKCR
-    Set-ItemProperty -Path "HKCR:\rldb\shell\open\command" -Name "(Default)" -Value ('"' + $PSScriptRoot + '\withdll.exe" "/d:' + $PSScriptRoot + '\GetSystemMetrics-Hook.dll" ' + $lockdown_runtime + ' "%1"')
+    if ($lockdown_installer.Name -like "LockDownBrowserOEMSetup.exe") {
+        # This is implmented *per OEM* so we need a custom implementation for each OEM
+        # The plan is to have another powershell script that will ask for the url and you can just paste it in.
```

Comment:
I wonder if we could use `AppInit_DLLs` for this maybe?

- https://learn.microsoft.com/en-us/windows/win32/win7appqual/appinit-dlls-in-windows-7-and-windows-server-2008-r2
- https://gist.github.com/hasherezade/9eedcf7176802966d652b3c9ba32d398